WebARonARKit/WebARonARKit.js

```diff
@@ -550,6 +550,41 @@
       return this.anchors_;
     };
 
+    // EventTarget implementation.
+    this.listeners = {};
+
+    this.addEventListener = function(type, callback) {
+      if (!(type in this.listeners)) {
+        this.listeners[type] = [];
+      }
+      this.listeners[type].push(callback);
+    };
+
+    this.removeEventListener = function(type, callback) {
+      if (!(type in this.listeners)) {
+        return;
+      }
+	  var stack = this.listeners[type];
+	  for (var i = 0, l = stack.length; i < l; i++) {
+	    if (stack[i] === callback){
+	      stack.splice(i, 1);
+	      return;
+	    }
+	  }
+	};
+
+    this.dispatchEvent = function(event) {
```

Comment:
If we want to be consistent as how events work in JS/DOM we should:
1) Allow to add functions and objects with handleEvent as callbacks.
2) Allow onXXX event handlers to be used too.

You can check the resize event handling in this same file.
I would check if Chromium eventtarget provides these functionalities out of the box.

Reply:
We aren't creating a standard class so I can't really inherit afaik. I wish I could, I don't love having to implement this ourselves.

I don't know what you mean by (1).
I'll try to get (2) to work.